File: notesV1/src/main/java/com/cnpeng/TempActivity.java
package com.cnpeng;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.cnpeng.cnpeng_demos2017_01.R;
import com.cnpeng.cnpeng_demos2017_01.databinding.ActivityTempBinding;
import com.cnpeng.utils.LogUtils;
import com.google.gson.Gson;

import org.apache.http.NameValuePair;
import org.apache.http.client.utils.URLEncodedUtils;

import java.net.URI;
import java.net.URISyntaxException;
import java.util.ArrayList;
import java.util.List;

import static com.cnpeng.utils.KotlinUtlisKt.isNetAvailable;


public class TempActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityTempBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_temp);

        if (isNetAvailable(this)){
            Toast.makeText(this, "调用 Kotlin 中扩展的函数判断网络是否可用", Toast.LENGTH_SHORT).show();
        }

        String str = getJsonStr();
        binding.tv.setText(str);


        try {
            URI query = new URI("ant://question/detail/open?detailID=7334&PUSHID=5b8f363a1d41c87e638a7c55");
            List<NameValuePair> list = URLEncodedUtils.parse(query, "UTF-8");
            if (null != list && !list.isEmpty()) {
                String value = list.get(list.size() - 1).getValue();
                LogUtils.d("解析出来的数据是", value);
            }
        } catch (URISyntaxException e) {
            e.printStackTrace();
        }
    }

    public String getJsonStr() {
        String str = "";
        List<Product> productList = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            productList.add(new Product("ID_00" + i, i));
        }

        Gson gson = new Gson();
        str = gson.toJson(productList);
        LogUtils.i("Json字符串", str);
        return str;
    }

    class Product {
        String id;
        int    num;

        public Product(String id, int num) {
            this.id = id;
            this.num = num;
        }
    }

}

File: notesV2/src/main/java/com/cnpeng/android2/d_mine/TempActivity.kt
package com.cnpeng.android2.d_mine

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.cnpeng.android2.R
import com.cnpeng.android2.databinding.ActivityTempBinding
import kotlinx.android.synthetic.main.activity_temp.*
import java.io.File

class TempActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding: ActivityTempBinding = DataBindingUtil.setContentView(this, R.layout.activity_temp)

        binding.tvTemp.text = "啥FUCK"

        initClickEvent()
    }

    private fun initClickEvent() {
        tv_getFile.setOnClickListener {

            val fileNames: MutableList<String> = mutableListOf()

            var fileNamesStr = ""

            val mPath = "/Users/cnpeng/CnPeng/00_LearningNotes"
            val fileTree: FileTreeWalk = File(mPath).walk()
            fileTree.maxDepth(8) //需遍历的目录层级为1时，表示无需检查子目录
                    .filter { it.isFile } //只挑选文件，不处理文件夹
                    .filter { it.extension == "md" } //选择扩展名为txt的文本文件
                    .forEach {
                        //循环处理符合条件的文件
                        fileNames.add(it.name)
                        fileNamesStr = "$fileNamesStr || ${it.name}"
                    }

            tv_temp.text = fileNamesStr
        }
    }
}





